### PR TITLE
Show usernames when hovering on player nickname in chat

### LINKED ISSUE #34: Show player’s actual name in chat when hovering on nickname
- No way to tell who actually sent a message if two players have the same nickname currently
- Will use ChatComponent

### PR DESCRIPTION
**Pull request by rfblock**
Resolves #34 
oh my god this was painful
i hope it works :crossed_fingers:

EDIT luke: Supersedes #96

### CODE REVIEW

**Comment by rfblock** [comment]
Should probably test on a private server first
Will probably just stop chatting or make double messages

**Comment by saharshy29** [comment]
Travis build is failing, but I can't seem to figure out why. @lukec11 can you take a look at this one.

File: src/main/java/com/hackclub/hccore/listeners/PlayerListener.java [hunk 1]
```diff
@@ -60,13 +60,11 @@ public void onAsyncPlayerChat(final AsyncPlayerChatEvent event) {
         net.md_5.bungee.api.ChatColor messageColor = data.getMessageColor().asBungee();
         net.md_5.bungee.api.ChatColor nameColor = data.getNameColor().asBungee();
         
-        BaseComponent[] replacementMessage = new ComponentBuilder(
-            event.getPlayer().getDisplayName()).color(nameColor).event(new HoverEvent(HoverEvent.Action.SHOW_TEXT, new ComponentBuilder(event.getPlayer().getName()).create()))
-            .append(" » ").color(ChatColor.GOLD.asBungee())
-            .append(ChatColor.translateAlternateColorCodes('&', event.getMessage())).color(messageColor)
-            .create();
+        BaseComponent nameComponent = new BaseComponent(event.getPlayer().getDisplayName()).setColor(nameColor).setHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, new ComponentBuilder(event.getPlayer().getName()).create()))
```
Comment (ifvictr) [inline src/main/java/com/hackclub/hccore/listeners/PlayerListener.java:63]: Missing semicolons

File: src/main/java/com/hackclub/hccore/listeners/PlayerListener.java [hunk 2]
```diff
@@ -60,13 +60,11 @@ public void onAsyncPlayerChat(final AsyncPlayerChatEvent event) {
...
+        BaseComponent arrowComponent = new BaseComponent(" » ").setColor(ChatColor.GOLD.asBungee())
```
Comment (ifvictr) [inline src/main/java/com/hackclub/hccore/listeners/PlayerListener.java:64]: Ditto

File: src/main/java/com/hackclub/hccore/listeners/PlayerListener.java [hunk 3]
```diff
@@ -60,13 +60,11 @@ public void onAsyncPlayerChat(final AsyncPlayerChatEvent event) {
...
+        BaseComponent playerChatComponent = new BaseComponent(ChatColor.translateAlternateColorCode('&', event.getMessage()).color(messageColor))
```
Comment (ifvictr) [inline src/main/java/com/hackclub/hccore/listeners/PlayerListener.java:65]: Ditto

**Comment by saharshy29** [comment]
Hmm. It seems that @lukec11 is very upset that I approved this pull request.

**Comment by jacobhaap** [comment]
hi Luke

**Comment by lukec11** [comment]
Code compiles & works correctly in testing - as long as styling is correct we can merge.

**Pull request merged**